### PR TITLE
Rewrite some tests not use `numpy_cupy_raises`

### PR DESCRIPTION
**Pull request by asi1024**
Ready for #3098.

### CODE REVIEW

File: tests/cupy_tests/sorting_tests/test_sort.py [hunk 1]
```diff
@@ -14,17 +14,19 @@ class TestSort(unittest.TestCase):
...
     # TODO(niboshi): Fix xfail
     @pytest.mark.xfail(reason='Explicit error types required')
-    @testing.numpy_cupy_raises()
-    def test_sort_zero_dim(self, xp):
-        a = testing.shaped_random((), xp)
-        a.sort()
+    def test_sort_zero_dim(self):
+        for xp in (numpy, cupy):
+            a = testing.shaped_random((), xp)
+            with pytest.raises(numpy.AxisError):
```
Comment (toslunar) [inline tests/cupy_tests/sorting_tests/test_sort.py]: Could you remove the involving `xfail`s in a new PR first?
- See https://github.com/cupy/cupy/pull/2870#discussion_r361290474 for the reason of `xfail`.
- CuPy `master` requires `numpy>=1.15`. (#2900)

File: tests/cupy_tests/core_tests/test_ndarray_copy_and_view.py [hunk 2]
```diff
@@ -39,17 +40,19 @@ def test_view_0d(self, xp):
...
     @testing.for_dtypes([numpy.int16, numpy.int64])
-    @testing.numpy_cupy_raises()
-    def test_view_non_contiguous_raise(self, xp, dtype):
-        a = testing.shaped_arange((2, 2, 2), xp, dtype=numpy.int32).transpose(
-            0, 2, 1)
-        a.view(dtype=dtype)
+    def test_view_non_contiguous_raise(self, dtype):
+        for xp in (numpy, cupy):
+            a = testing.shaped_arange((2, 2, 2), xp, dtype=numpy.int32)
+            a = a.transpose(0, 2, 1)
```
Comment (toslunar) [inline tests/cupy_tests/core_tests/test_ndarray_copy_and_view.py]: Note: I approve this refactoring.

File: tests/cupy_tests/core_tests/test_ndarray.py [hunk 3]
```diff
@@ -373,32 +376,34 @@ def test_axis_overrun2(self):
...
+            i = testing.shaped_arange(self.indices, xp, numpy.int32) % 3
+            o = testing.shaped_arange(self.out_shape, xp)
+            with pytest.raises(ValueError):
+                wrap_take(a, i, out=o)
 
 
 @testing.parameterize(
     {'shape': (3, 4, 5), 'indices': (2, 3), 'out_shape': (2, 3)},
-    {'shape': (), 'indices': 0, 'out_shape': ()}
+    {'shape': (), 'indices': (), 'out_shape': ()}
```
[the remark this answers is not in the document]
Comment (asi1024) [reply tests/cupy_tests/core_tests/test_ndarray.py]: In the original version, a `TypeError` is raised in `testing.shaped_arange`.

File: tests/cupy_tests/sorting_tests/test_sort.py (hunk 1 above)
Comment (asi1024) [reply tests/cupy_tests/sorting_tests/test_sort.py]: Splitted into #3167.

File: tests/cupy_tests/core_tests/test_ndarray.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (toslunar) [reply tests/cupy_tests/core_tests/test_ndarray.py]: Thanks.  It's possible that `indices=(0,)` was intended.

**Comment by asi1024** [comment]
@toslunar PTAL.

File: tests/cupy_tests/core_tests/test_ndarray.py [hunk 4]
```diff
@@ -373,32 +376,34 @@ def test_axis_overrun2(self):
...
 class TestNdarrayTakeErrorShapeMismatch(unittest.TestCase):
 
-    @testing.numpy_cupy_raises()
-    def test_shape_mismatch(self, xp):
-        a = testing.shaped_arange(self.shape, xp)
-        i = testing.shaped_arange(self.indices, xp, numpy.int32) % 3
-        o = testing.shaped_arange(self.out_shape, xp)
-        wrap_take(a, i, out=o)
+    def test_shape_mismatch(self):
+        for xp in (numpy, numpy):
```
Comment (toslunar) [inline tests/cupy_tests/core_tests/test_ndarray.py]: Typos here and in #3167.
```
(git)[5562f3a45]% git grep --line-number '(numpy, numpy)'
tests/cupy_tests/core_tests/test_ndarray.py:385:        for xp in (numpy, numpy):
tests/cupy_tests/sorting_tests/test_sort.py:22:        for xp in (numpy, numpy):
```

Comment (asi1024) [reply tests/cupy_tests/core_tests/test_ndarray.py]: Fixed.

**Comment by toslunar** [comment]
Jenkins, test this please.

File: tests/cupy_tests/core_tests/test_ndarray_adv_indexing.py [hunk 5]
```diff
@@ -326,10 +330,11 @@ def test_invalid_adv_getitem(self, xp):
...
 
-    @testing.numpy_cupy_raises()
-    def test_invalid_adv_getitem(self, xp):
-        a = testing.shaped_arange(self.shape, xp)
-        a[self.indexes]
+    def test_invalid_adv_getitem(self):
+        for xp in (numpy, cupy):
+            a = testing.shaped_arange(self.shape, xp)
+            with pytest.raises(IndexError):
+                a[self.indexes]
```
Comment (toslunar) [inline tests/cupy_tests/core_tests/test_ndarray_adv_indexing.py]: The test failed on Jenkins.

Comment (asi1024) [reply tests/cupy_tests/core_tests/test_ndarray_adv_indexing.py]: Removed this test as it is for the old NumPy version.

**Comment by toslunar** [comment]
Jenkins, test this please.

**Pull request merged**